packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -2,26 +2,38 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:meta/meta.dart';
 import 'package:pool/pool.dart';
 
 import '../../asset.dart';
 import '../../base/file_system.dart';
+import '../../base/logger.dart';
+import '../../build_info.dart';
+import '../../convert.dart';
 import '../../devfs.dart';
 import '../../globals.dart' as globals;
 import '../build_system.dart';
 import '../depfile.dart';
 import 'dart.dart';
 import 'icon_tree_shaker.dart';
 
+/// The input key for an SkSL bundle path.
+const String kBundleSkSLPath = 'BundleSkSLPath';
+
 /// A helper function to copy an asset bundle into an [environment]'s output
 /// directory.
 ///
 /// Throws [Exception] if [AssetBundle.build] returns a non-zero exit code.
 ///
+/// [skSLBundle] may optionally contain a validated SkSL shader bundle.
+///
 /// Returns a [Depfile] containing all assets used in the build.
-Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) async {
+Future<Depfile> copyAssets(Environment environment, Directory outputDirectory, {
+  Map<String, DevFSContent> additionalContent,
+}) async {
   final File pubspecFile =  environment.projectDir.childFile('pubspec.yaml');
-  final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
+  final ManifestAssetBundle assetBundle = AssetBundleFactory.instance.createBundle()
+    as ManifestAssetBundle;
```

Comment:
I don't love this cast, I think we get way too cute with factory patterns in flutter_tools--sometimes it's better to just make what you need if you need to understand the internal details of it anyway.

Reply:
Done!